### PR TITLE
Security layer 2 for Bifold

### PR DESCRIPTION
**Pull request by MosCD3**
Signed-off-by: Mostafa <mostafa.youssef@ontario.ca>

# Summary of Changes

This PR will be followed up with some cleanup and contains the following changes:
- Updated security service 
- Biometrics flags added
- Wired up the wallet with the security service through an Auth provider hooks
- Wallet id/key now derives from user created PIN and stored in secure keychain 
- Wallet init by using the Auth hooks to get the wallet secret from keychain

Still to be done and added in a followup:
- Re-opening the app after first login to compare pin using either Biometry+PIN or Biometry+Fallback to PIN or .. etc
- Adding security level enum to environment config or settings screen

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [ ] Updated documentation as needed for changed code and new or modified features;
- [ ] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Comment by amanji** [comment]
Is this PR ready to merge? If so can we clean up the PR description? Is there anything in progress or is it all complete?

File: core/App/constants.ts [hunk 1]
```diff
@@ -12,8 +12,10 @@ export enum LocalStorageKeys {
 }
 
 //Keys for items saved in keychain/async storage
-export const KEYCHAIN_SERVICE_KEY = 'walletkey'
+export const KEYCHAIN_SERVICE_KEY = 'secret.wallet.key'
 export const KEYCHAIN_SERVICE_ID = 'walletid'
+export const KEYCHAIN_SERVICE_PIN_KEY = 'secret.pin.key'
+export const KEYCHAIN_SERVICE_RAND_KEY = 'secret.rand.key'
 export const STORAGE_KEY_SALT = 'savedalt'
```
Comment (amanji) [inline core/App/constants.ts]: Should be 'savedsalt'?

File: core/App/services/kdf.service.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/*
```
Comment (amanji) [inline core/App/services/kdf.service.ts]: I think we can clean up this comment

File: core/App/screens/PinEnter.tsx [hunk 3]
```diff
@@ -26,15 +34,70 @@ const PinEnter: React.FC<PinEnterProps> = ({ setAuthenticated }) => {
...
+        return
+      } else {
+        setAuthenticated(true)
+      }
+    }
+  }
+
+  const initWithBiometrics = async () => {
+    // TODO: get auth level from settings
+    // const authLevel = await getAuthLevel()
```
Comment (amanji) [inline core/App/screens/PinEnter.tsx:66]: Will this be required in the future?

**Review by jleach — DISMISSED** [review]
Looks good. A bit of cleanup in a subsequent PR would be great but otherwise LGTM.

File: core/App/screens/PinEnter.tsx (hunk 3 above)
Comment (MosCD3) [reply core/App/screens/PinEnter.tsx:66]: Yes, it is hardcoded now but next level it should be configured in the settings or at least an environment variable

File: core/App/services/kdf.service.ts (hunk 2 above)
Comment (MosCD3) [reply core/App/services/kdf.service.ts]: done

File: core/App/constants.ts (hunk 1 above)
Comment (MosCD3) [reply core/App/constants.ts]: done

**Review by amanji — APPROVED** [review]

**Pull request merged**